source/vstd/pervasive.rs

```diff
@@ -412,3 +412,31 @@ pub open spec fn cloned<T: Clone>(a: T, b: T) -> bool {
 }
 
 } // verus!
+
+verus! {
+
+#[verifier(external_body)]
+pub fn call_vpanic(_out: &[&str]) -> ! {
+    unimplemented!()
+}
+
+#[verifier(external_body)]
+pub fn vpanic_fmt<T: core::fmt::Debug>(_v: &T) -> &str {
+    unimplemented!()
+}
```

Comment:
Adding panic functions to `vstd` will affect https://github.com/verus-lang/verus/discussions/1292 , because calling `call_vpanic` could be considered a form of "cheating" if we're trying to prove that the code never panics, which we are by default (cc @utaal ).  For now, should these two functions be guarded by `#[cfg(feature = "vpanic")]`?

Reply:
How could we prove our code never panics in current verus? If we want to ensure never panic, I can add a requires guard to the call_vpanic. If my code uses panic but it is unreachable, it should still be a correct implementation that never panics.

fn call_vpanic() -> ! 
requires
   !spec_never_panic()
{}

Reply:
> How could we prove our code never panics in current verus?

Currently Verus's design and library wrappers all add preconditions to operations that might panic (e.g., dividing by zero, or indexing into an array).  If we wanted to add support for code that contains `panic!` (ignoring the issues you raise with trying to do that at present), we could add `requires false` as a precondition, which would oblige Verus to prove that the call is always unreachable.

Reply:
Verus already allows you to specify if a function can panic or not.

https://verus-lang.github.io/verus/guide/reference-unwind-sig.html

Reply:
@tjhance, that is what I want to ask about. My prior understanding is that developers need to specify `no_unwind when true` to prove there is no panic. My call_vpanic does not specify no_unwind and so it should be `no_unwind when false`. Then it should not do "cheating" if we want to prove a function with panic!() call never panic. But I thought there might be a flag or some mode that verus can use to set no_unwind when true as default. Or I should just explicitly add `no_unwind when false`